utils/obsproc/IcecMirs2Ioda.h

```diff
@@ -0,0 +1,165 @@
+      gdasapp::obsproc::iodavars::IodaVars iodaVars(nobs, floatMetadataNames, intMetadataNames);
+
+      oops::Log::debug() << "--- iodaVars.location_: " << iodaVars.location_ << std::endl;
+
+      // Read non-optional metadata: longitude and latitude
+      std::vector<float> oneDimLatVal(iodaVars.location_);
+      ncFile.getVar("Latitude").getVar(oneDimLatVal.data());
+
+      std::vector<float> oneDimLonVal(iodaVars.location_);
+      ncFile.getVar("Longitude").getVar(oneDimLonVal.data());
+
+      // Create a vector to hold the full Qc variable
+      std::vector<int> fullQcVar(nqcs);
+      ncFile.getVar("Qc").getVar(fullQcVar.data());
+
+      // Read Quality Flags as a preQc
+      // Create a vector to hold the first slice
+      std::vector<int> oneDimFlagsVal(nobs);
+      // Extract the first slice
+      for (int i = 0; i < nobs; ++i) {
+          oneDimFlagsVal[i] = fullQcVar[i * dimQcSize];
+      }
+
+      // Get Ice_Concentration obs values
+      std::vector<int16_t> oneDimObsVal(iodaVars.location_);
+      ncFile.getVar("SIce").getVar(oneDimObsVal.data());
+
+      // Read and process the dateTime
+      std::vector<int64_t> TimeYearVal(dimxSize);
+      ncFile.getVar("ScanTime_year").getVar(TimeYearVal.data());
+
+      std::vector<int64_t> TimeMonthVal(dimxSize);
+      ncFile.getVar("ScanTime_month").getVar(TimeMonthVal.data());
+
+      std::vector<int64_t> TimeDayVal(dimxSize);
+      ncFile.getVar("ScanTime_dom").getVar(TimeDayVal.data());
+
+      std::vector<int64_t> TimeHourVal(dimxSize);
+      ncFile.getVar("ScanTime_hour").getVar(TimeHourVal.data());
+
+      std::vector<int64_t> TimeMinuteVal(dimxSize);
+      ncFile.getVar("ScanTime_minute").getVar(TimeMinuteVal.data());
+
+      std::vector<int64_t> TimeSecondVal(dimxSize);
+      ncFile.getVar("ScanTime_second").getVar(TimeSecondVal.data());
+
+      iodaVars.referenceDate_ = "seconds since 1970-01-01T00:00:00Z";
+
+      for (int i = 0; i < dimxSize; i++) {
+        int year = TimeYearVal[i];
+        int month = TimeMonthVal[i];
+        int day = TimeDayVal[i];
+        int hour = TimeHourVal[i];
+        int minute =  TimeMinuteVal[i];
+        int second = TimeSecondVal[i];
+
+        // Avoid crash util in ioda::convertDtimeToTimeOffsets
+        if (year == -999 || month == -999 || day == -999 ||
+          hour == -999 || minute == -999 || second == -999) {
+          year = month = day = hour = minute = second = 0;
+        }
+
+        // Construct iso8601 string format for each dateTime
+        std::stringstream ss;
+        ss << std::setfill('0')
+           << std::setw(4) << year << '-'
+           << std::setw(2) << month << '-'
+           << std::setw(2) << day << 'T'
+           << std::setw(2) << hour << ':'
+           << std::setw(2) << minute << ':'
+           << std::setw(2) << second << 'Z';
+        std::string formattedDateTime = ss.str();
+        util::DateTime dateTime(formattedDateTime);

```

Comment:
I wonder if there's a utility for this

Reply:
IODA util has it- We have used it to another converters, too